src/sql/orderbook/prod_order_rewards.sql

```diff
@@ -33,293 +33,148 @@ with trade_hashes as (
         settlement.block_number > {{start_block}}
         and settlement.block_number <= {{end_block}}
 ),
-order_surplus AS (
+-- order data
+order_data AS (
     SELECT
-        ss.winner as solver,
+        uid,
+        sell_token,
+        buy_token,
+        sell_amount,
+        buy_amount,
+        kind,
+        app_data
+    FROM orders
+    UNION ALL
+    SELECT
+        uid,
+        sell_token,
+        buy_token,
+        sell_amount,
+        buy_amount,
+        kind,
+        app_data
+    FROM jit_orders
+),
+protocol_fee_policies_application_order AS (
+    SELECT
+        fp.auction_id,
+        fp.order_uid,
+        MIN(fp.application_order) as application_order
+    FROM fee_policies fp JOIN trade_hashes th ON fp.auction_id = th.auction_id AND fp.order_uid = th.order_uid
+    GROUP BY fp.auction_id, fp.order_uid
+),
+protocol_fee_kind AS (
+    SELECT
+        fp.auction_id,
+        fp.order_uid,
+        fp.kind
+    FROM fee_policies fp JOIN protocol_fee_policies_application_order pfpao ON fp.auction_id = pfpao.auction_id AND fp.order_uid = pfpao.order_uid AND fp.application_order = pfpao.application_order
+),
```

Comment:
I am a bit uneasy about introducing changes here which are not part of the solver-rewards query. But since we plan to revamp some of them, this should be fine for some time.

An alternative implementation would be
```suggestion
protocol_fee_kind AS (
    SELECT DISTINCT ON (fp.auction_id, fp.order_uid)
        fp.auction_id,
        fp.order_uid,
        fp.kind
    FROM fee_policies fp
        JOIN trade_hashes th
        ON fp.auction_id = th.auction_id AND fp.order_uid = th.order_uid
    ORDER BY fp.auction_id, fp.order_uid, fp.application_order
),
```

Reply:
Note that this part actually changes also the semantics of the protocol_fee_kind entry on Dune, which now always refers to the protocol fee kind and NOT the partner fee kind associated with the order.

I guess something a bit more explicit that would be easier for me to read would be to add the `limit 1` suffix. But what you suggest also is fine.